sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -93,19 +92,18 @@ private static void CompareActualToExpected(IEnumerable<Diagnostic> diagnostics,
 
                 foreach (var secondaryLocation in secondaryLocations)
                 {
-                    VerifyIssue(expectedIssues,
+                    VerifySecondaryIssue(expectedIssues,
                         issue => issue.IssueId == issueId && !issue.IsPrimary,
                         secondaryLocation.Location,
                         secondaryLocation.Message,
-                        null,
-                        out issueId);
+                        issueId);
                 }
             }
 
             if (expectedIssues.Count != 0)
             {
-                Execute.Assertion.FailWith($"Issue expected but not raised on line(s) " +
-                    $"{string.Join(",", expectedIssues.Select(i => i.LineNumber))}.");
+                var expectedIssuesDescription = expectedIssues.Select(i => $"{Environment.NewLine}Line: {i.LineNumber}, Type: {IssueType(i.IsPrimary)}, Id: '{i.IssueId}'");
+                Execute.Assertion.FailWith($"Issue(s) expected but not raised on line(s):{expectedIssuesDescription.JoinStr("")}");
```

Comment:
Creating empty strings is not good practice for memory reasons. null is cheap and will work just as good.
```suggestion
                Execute.Assertion.FailWith($"Issue(s) expected but not raised on line(s):{expectedIssuesDescription.JoinStr(null)}");
```

Reply:
This should actually not create any new empty string, as empty literals will reuse the interned empty string.
I'd much prefer to keep `expectedIssuesDescription.JoinStr("")` as I find it much more explicit that we are joining using an empty string

Reply:
The interning of he empty string is what I wanted to avoid.
null and "" looks the same to me when the overall result cannot be null.